TT-1624 Matching user based on surname and date of birth

SKeerthana opened the pull request.

Commit:
TT-1624 - Matching users based on surname

Authors: @SKeerthana @sgreensmith

Commit:
TT-1624 - Matching users based on verified date of birth

Changing joda time to java time since jdbi does not support
joda time out of the box

Authors: @SKeerthana @sgreensmith

SKeerthana commented on src/main/java/uk/gov/ida/verifylocalmatchingserviceexample/model/Person.java:
```diff
@@ -0,0 +1,36 @@
+    public void setSurname(String surname) {
+        this.surname = surname;
+    }
+
```
setters are required by JDBI. If someone knows how we can avoid, it would be great!

willp-bl replied:
might be worth looking at this branch of stub-idp where I used JDBI, don't think that needed setters... https://github.com/alphagov/ida-stub-idp/compare/bau/use_jdbi

SKeerthana replied:
I will have a look at it. Thanks!

georgievh commented on src/main/java/uk/gov/ida/verifylocalmatchingserviceexample/VerifyLocalMatchingServiceExampleApplication.java:
```diff
@@ -43,11 +44,8 @@ public void run(VerifyLocalMatchingServiceExampleConfiguration configuration, En
-        Cycle0MatchingService cycle0MatchingService = new Cycle0MatchingService(verifiedPid);
-        environment.jersey().register(new MatchingServiceResource(cycle0MatchingService));
+        MatchingService matchingService = factory.getMatchingService(configuration.getDatabaseConfiguration().getUrl());
+        environment.jersey().register(new MatchingServiceResource(matchingService));
```
We should probably move all new Object statements to the factory

Commit:
TT-1624 - Moving resource construction into factory method

Authors: @SKeerthana @sgreensmith

Commit:
TT-1624 - Adding jdbi row mapper for person class

-JDBI is not able to map sqlite date's to java LocalDate's
but it works fine for h2 in-memory database
-And also introdusing mappers avoid setters in Person class

Authors: @SKeerthana @sgreensmith
